content/en/docs/concepts/containers/container-lifecycle-hooks.md

```diff
@@ -64,10 +64,20 @@ the Container ENTRYPOINT and hook fire asynchronously.
 However, if the hook takes too long to run or hangs,
 the Container cannot reach a `running` state.
 
-The behavior is similar for a `PreStop` hook.
-If the hook hangs during execution,
-the Pod phase stays in a `Terminating` state and is killed after `terminationGracePeriodSeconds` of pod ends.
-If a `PostStart` or `PreStop` hook fails,
+`PreStop` hooks are not executed asynchronously from the signal
+to stop the Container; the hook must complete its execution before
+the signal can be sent.
+If a `PreStop` hook hangs during execution,
+the Pod phase stays in a `Terminating` state and is killed after its `terminationGracePeriodSeconds` expires.
```

Comment:
Hello @2rs2ts . Thanks for opening this pull request. I agree that this paragraph is hard to follow.

I am not completely clear on this sentence:
`PreStop` hooks are not executed asynchronously from the signal
to stop the container.

Is the sentence equivalent to:
`PreStop` hooks and the signal to stop the container are not executed asynchronously.
Each `PreStop` hook must complete before the signal can be sent to stop the container.

What do you think about removing the word "phase"?
If a `PreStop` hook hangs during execution,
the Pod stays in a `Terminating` state and ...
[what?] is destroyed after its `terminationGracePeriodSeconds` expires.

Reply:
As you might have seen in my PR summary the problem with this section from before was that it implied that `PreStop` hooks were asynchronous because it described `PostStart` hooks as such and then said `PreStop` hooks were much the same. So this is meant to be explicit. If you like we can try to reword this whole section so that it's very very clear that `PostStart` is *not* to `ENTRYPOINT` as `PreStop` is to a signal to stop the container.

I do not think removing the word "phase" is good but I can change how I use it. It is quite literally the Pod's `phase` so I think it is good to use the exact word for it, no?